File: backend/repository/src/main/java/ru/relex/aim/repository/entity/RequestStatus.java
package ru.relex.aim.repository.entity;

/**
 * Represents a stage of request handling.
 *
 * @author Sorokin Georgy
 */
public enum RequestStatus {

  NEW(1, "Новая"),
  IN_PROGRESS(2, "В работе"),
  EXECUTED(3, "Исполнена"),
  CANCELED(4, "Отменена"),
  PAUSED(5, "Приостановлена");

  private final int id;
  private final String name;

  RequestStatus(int id, String name) {
    this.id = id;
    this.name = name;
  }

  public String getName() {
    return name;
  }

  public int getId() {
    return id;
  }

  /**
   * Tries to parse given id into {@link RequestStatus} instance.
   *
   * @param id id to find matching request status
   * @return request status with matching id or {@literal null}.
   */
  public static RequestStatus fromId(final int id) {
    for (final var value : values()) {
      if (value.id == id) {
        return value;
      }
    }
    return null;
  }
}

File: frontend/src/model/request-status.ts
import { RequestStatus } from './get/request-status';

export function RequestStatuses(t: (s: string) => string) {
  return [
    {
      key: RequestStatus.NEW.toString(),
      value: t('requestPage.status.new'),
    },
    {
      key: RequestStatus.IN_PROGRESS.toString(),
      value: t('requestPage.status.inWork'),
    },
    {
      key: RequestStatus.PROCESSED.toString(),
      value: t('requestPage.status.executed'),
    },
    {
      key: RequestStatus.CANCELED.toString(),
      value: t('requestPage.status.canceled'),
    },
    {
      key: RequestStatus.PAUSED.toString(),
      value: t('requestPage.status.paused'),
    },
  ];
}
